src/main/java/com/aws/greengrass/util/platforms/unix/linux/CgroupSubSystemV2.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.unix.linux;
+
+public enum CgroupSubSystemV2 {
+    Memory("", ""), CPU("", ""), Freezer("", ""),
+    Unified("","");
+
+    private String osString;
+    private String mountSrc;
+
+    CgroupSubSystemV2(String osString, String mountSrc) {
```

Comment:
getComponentMemoryLimitPath and those similar methods seem like they should be implemented here instead of checking for an empty string in the Cgroup class.